src/server/session.ts

```diff
@@ -1,9 +1,19 @@
+/// <reference types="node" />

```

Comment:
What types do you need for this? I see you are defining the 'net' module types inline below.

Reply:
The reference is needed for `require` and `Buffer` to be defined, used to import `"net"` and write events to the socket, respectively.

Reply:
my 2 cents: `session.ts` used to be a abstracted from any host specific bits so it can be used in any environment + it was also very convenient for testing since anything can be mocked. All host specific bits were injected from the outside in `iosession.ts`.